scripts/build/termux_download.sh

```diff
@@ -1,53 +1,68 @@
 #!/usr/bin/bash
 
 termux_download() {
-	if [ $# != 3 ]; then
+	if [[ $# != 3 ]]; then
 		termux_error_exit "termux_download(): Invalid arguments - expected \$URL \$DESTINATION \$CHECKSUM"
 	fi
 	local URL="$1"
 	local DESTINATION="$2"
 	local CHECKSUM="$3"
 
-	if [ -f "$DESTINATION" ] && [ "$CHECKSUM" != "SKIP_CHECKSUM" ]; then
+	if [[ -f "$DESTINATION" ]] && [[ "$CHECKSUM" != "SKIP_CHECKSUM" ]]; then
 		# Keep existing file if checksum matches.
 		local EXISTING_CHECKSUM
-		EXISTING_CHECKSUM=$(sha256sum "$DESTINATION" | cut -f 1 -d ' ')
-		if [ "$EXISTING_CHECKSUM" = "$CHECKSUM" ]; then return; fi
+		EXISTING_CHECKSUM=$(sha256sum "$DESTINATION" | cut -d' ' -f1)
+		[[ "$EXISTING_CHECKSUM" == "$CHECKSUM" ]] && return
 	fi
 
 	local TMPFILE
 	local -a CURL_OPTIONS=(
 		--fail
-		--retry 20
+		--retry 10
 		--retry-connrefused
-		--retry-delay 30
+		--retry-delay 5
+		--speed-limit 10000
```

Comment:
To clarify, this is a lower bound.
Not an upper bound?

Reply:
Yes, it is a lower bound. And would be better to change it to `1000` bytes, `curl` uses `1` byte by default if `--speed-time` is passed without `--speed-limit`. I am pretty sure `1000` bytes/`1KB` could happen for some very slow source repos and should still indicate a working download connection.

https://manpages.debian.org/testing/curl/curl.1.en.html#Y,